### PR TITLE
MSYS2 fixes

### PR DESCRIPTION
**Pull request by nulano**
Changes proposed in this pull request:

 * Remove old include directory `C:\\msys64\\mingw32\\include\\libimagequant`.

   It uses a hardcoded path to MSYS2, it is hardcoded to 32-bit, and it is no longer necessary, as the include is now in `C:\\msys64\\mingw32\\include\\libimagequant.h`.

 * Remove GCC workaround for MSYS, see https://github.com/python-pillow/Pillow/issues/4796#issuecomment-683370940

   > The issue was that AppVeyor was missing the MSYSTEM=MINGW32 envvar, and getting confused by the default MSYSTEM=MSYS2.

 * Fix LibTiff support. The issue is that MSYS2 builds LibTiff with `USE_WIN32_FILEIO`, which changes the meaning of the `fd` parameter for `TIFFFdOpen`, see https://github.com/python-pillow/Pillow/issues/4237#issuecomment-569403417. I've added a workaround by converting the parameter before passing it to LibTiff if building on MinGW.

   It might be possible to detect the system version of LibTiff by including the `tif_config.h` header to fix&nbsp;#4237 (by enabling the workaround if `USE_WIN32_FILEIO` was enabled), but it appears to be a private header (while it is available in MSYS2, it is not available in Ubuntu). See also [passing build using CMake on Windows](https://github.com/nulano/Pillow/runs/1046684238) from my branch https://github.com/nulano/Pillow/compare/mingw-setup...nulano:tiff-test.

   Another possible fix would involve absorbing `TIFFFdOpen` from `tif_unix.c` into Pillow to avoid this issue by using the stable `TiffClientOpen`.

 * [Document installation on MSYS2](https://pillow--4890.org.readthedocs.build/en/4890/installation.html#building-on-windows-using-msys2-mingw)

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -299,6 +301,42 @@ If you wish to compile Pillow manually, you can use the build scripts
...
+    pacman -S \
+        mingw-w64-x86_64-libjpeg-turbo \
+        mingw-w64-x86_64-zlib \
+        mingw-w64-x86_64-libtiff \
+        mingw-w64-x86_64-freetype \
+        mingw-w64-x86_64-lcms2 \
+        mingw-w64-x86_64-libwebp \
+        mingw-w64-x86_64-openjpeg2 \
+        mingw-w64-x86_64-libimagequant \
+        mingw-w64-x86_64-libraqm
```
Comment (radarhere) [inline docs/installation.rst:332]:
```suggestion
        mingw-w64-x86_64-freetype \
        mingw-w64-x86_64-lcms2 \
        mingw-w64-x86_64-libimagequant \
        mingw-w64-x86_64-libjpeg-turbo \
        mingw-w64-x86_64-libraqm
        mingw-w64-x86_64-libtiff \
        mingw-w64-x86_64-libwebp \
        mingw-w64-x86_64-openjpeg2 \
        mingw-w64-x86_64-zlib \
```
Sorted

Comment (nulano) [reply docs/installation.rst:332]: The order is the same as in the External Libraries section above. I think it is more helpful to follow that order for easier cross-referencing if someone doesn't want a full install.

**Comment by lazka** [comment]
You could use https://github.com/marketplace/actions/setup-msys2

**Comment by hugovk** [comment]
Thanks!

**Pull request merged**